clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1871,4 +1873,100 @@ void handleFailure(Throwable throwable) {
         return new CreatePartitionsResult(new HashMap<String, KafkaFuture<Void>>(futures));
     }
 
+    public DeleteRecordsResult deleteRecords(final Map<TopicPartition, RecordsToDelete> partitionsAndOffsets,
+                                             final DeleteRecordsOptions options) {
+
+        // requests need to be sent to partitions leader nodes so ...
+        // ... from the provided map it's needed to create more maps grouping topic/partition per leader
+
+        final Map<TopicPartition, KafkaFutureImpl<Long>> futures = new HashMap<>(partitionsAndOffsets.size());
+        for (TopicPartition topicPartition: partitionsAndOffsets.keySet()) {
+            futures.put(topicPartition, new KafkaFutureImpl<Long>());
+        }
+
+        // preparing topics list for asking metadata about them
+        final List<String> topics = new ArrayList<>();
```

Comment:
Should Set be used here (since there may be duplicate topics from the loop below) ?

Reply:
You are right. I have just fixed that. Thanks !